lib/mutant/expression/namespace.rb

```diff
@@ -2,24 +2,12 @@ module Mutant
   class Expression
     # Abstract base class for expressions matching namespaces
     class Namespace < self
-      include AbstractType
-
-    private
-
-      # Return matched namespace
-      #
-      # @return [String]
-      #
-      # @api private
-      #
-      def namespace
-        match[__method__]
-      end
+      include AbstractType, Anima.new(:scope_name)
+      private(*anima.attribute_names)
 
       # Recursive namespace expression
       class Recursive < self
-
-        register(/\A(?<namespace>#{SCOPE_PATTERN})?\*\z/)
+        REGEXP = /\A(?:#{SCOPE_NAME_PATTERN})?\*\z/.freeze
```

Comment:
What about defining `SCOPE_NAME_PATTERN` so it is wrapped in `(?:` and `)` ? Then it can be treated as an opaque matching pattern without knowing what it's contents are. You can then just use `#{SCOPE_NAME_PATTERN}?`.